core/descriptors.jvm/src/org/jetbrains/kotlin/load/java/lazy/types/JavaTypeResolver.kt

```diff
@@ -69,13 +70,13 @@ class JavaTypeResolver(
         val annotations = LazyJavaAnnotations(c, arrayType, areAnnotationsFreshlySupported = true)
 
         if (primitiveType != null) {
-            val jetType = c.module.builtIns.getPrimitiveArrayKotlinType(primitiveType)
-
-            jetType.replaceAnnotations(Annotations.create(annotations + jetType.annotations))
+            val kotlinType = c.module.builtIns.getPrimitiveArrayKotlinType(primitiveType).let {
```

Comment:
Are changes in this file just a code cleanup, or you expect some change of semantics here?

Reply:
This is the main part of the fix - `replaceAnnotations` returns a new type, it does not modify the existing one.